contributors/design-proposals/volume-user-configuration.md

```diff
@@ -0,0 +1,200 @@
+* *regular user* - user that can create PersistentVolumeClaims and cannot edit StorageClasses.
+
+## Use cases
+
+* Regular user wants get a PV of storage class "fast" in availability zone "east".
+* Admin wants to allow users to create PVs with storage class "fast" only in availability zones "east" and "north".
+* Regular user wants to discover what tunable properties for storage class "fast" are available and as result he can see that parameter "availability zone" with allowed values "east" and "north" is available.
+
+While availability zone is used as example in use cases, provisioners can choose any property such as encryption key, IOPS or replica count as tunable.
+
+## Design
+
+* Parameters that are tunable by regular users are implemented as label selector in `pvc.spec.selector`. We already use labels for zones and using a label selector for provisioning was already discussed in [storage class proposal](volume-provisioning.md).
+* All provisioners (both internal and external) will allow both `pvc.spec.selector` and `pvc.spec.class` to be set. The provisioners *must* satisfy both these conditions during provisioning, i.e. they must provision something that has labels required by the selector and all properties of the class. If this combination is not possible to achieve, the provisioner must signal an error to the user in usual way (i.e. via an event on the PVC).
+  * In other words, if the selector requires a PV to have labels `"zone" not in ["east", "south"]`, the provisioner must understand what "zone" means, what zones are available and provision something e.g. in zone "north". 
+  * The provisioner is required to validate the pvc.spec.selector and refuse any labels that it does not understand. Provisioners may choose not to allow any selector to be setif they don't support any parameters to be tunable by regular users.
+
+* StorageClass will get a new field to tell regular users what are allowed labels in pvc.spec.selector:
+    ```
+    type StorageClass struct {
+        // <snip>
+
+        // selectorOptions contains details about labels that are allowed by this
+        // StorageClass in selectors of PersistentVolumeClaims.
+        SelectorOptions []SelectorComponent
+    }
+
+    // SelectorComponent describes one label that can be used in PVC selector
+    // when requesting a volume of a particular StorageClass
+    type SelectorComponent struct {
+        // Label selector key.
+        Key string
+
+        // Optional default selector value if it is unspecified in the PVC.
+        DefaultValue *string
+
+        // Optional flag whether this selector key is mandatory in the PVC.
+        // Defaults to "false" when missing.
+        Required *bool
+
+        // Optional list of values that are allowed for this label selector key.
+        AllowedValues []string
+
+        // Optional display name. May not contain newlines. If unspecified, key
+        // should be used.
+        DisplayName *string
```

Comment:
remove the human-readable fields displayName/description, since there were concerns about translation, etc